src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -513,6 +524,65 @@ def getThresholds(self):
 
         return
 
+    def _getJobSubmitCondition(self, jobPrio, siteName, jobType):
+        """
+        returns the string whehter job is read to submit or the reason can't be submitted
+        """
+        try:
+            totalPendingSlots = self.currentRcThresholds[siteName]["total_pending_slots"]
+            totalPendingJobs = self.currentRcThresholds[siteName]["total_pending_jobs"]
+            totalRunningSlots = self.currentRcThresholds[siteName]["total_running_slots"]
+            totalRunningJobs = self.currentRcThresholds[siteName]["total_running_jobs"]
+            highestPriorityInJobs = self.currentRcThresholds[siteName]['wf_highest_priority']
+
+            taskPendingSlots = self.currentRcThresholds[siteName]['thresholds'][jobType]["pending_slots"]
+            taskPendingJobs = self.currentRcThresholds[siteName]['thresholds'][jobType]["task_pending_jobs"]
+            taskRunningSlots = self.currentRcThresholds[siteName]['thresholds'][jobType]["max_slots"]
+            taskRunningJobs = self.currentRcThresholds[siteName]['thresholds'][jobType]["task_running_jobs"]
+
+            self.currentRcThresholds[siteName].setdefault("init_total_pending_slots", totalPendingSlots)
+            self.currentRcThresholds[siteName]['thresholds'][jobType].setdefault("init_task_pending_slots", taskPendingSlots)
+            initTotalPending = self.currentRcThresholds[siteName]["init_total_pending_slots"]
+            initTaskPending = self.currentRcThresholds[siteName]['thresholds'][jobType]["init_task_pending_slots"]
+
+        except KeyError as ex:
+            msg = "Invalid key for site %s and job type %s\n" % (siteName, jobType)
+            msg += traceback.format_exc()
+            logging.error(msg)
+            return "NoJobType_%s_%s" % (siteName, jobType)
+
+        if highestPriorityInJobs is None or jobPrio <= highestPriorityInJobs:
+            # there is no pending or running jobs in the system (None case) or
+            # priority of the job is lower or equal don't allow overflow
+            totalPendingThreshold = totalPendingSlots
+            taskPendingThreshold = taskPendingSlots
+            totalJobThreshold = totalPendingSlots + totalRunningSlots
+            totalTaskTheshold = taskPendingSlots + taskRunningSlots
+        else:
+            # priorty of the job is higher than any off currently pending or running.
+            # increase the threshold by condorOverflowFraction * origianl pending slot.
+            pendingTotalOverflow = max(totalPendingSlots, initTotalPending) + (
```

Comment:
the reason I did max(totalPendingSlots, initTotalPending) to handle when 
totalPendingSlots >  initTotalPending. Lets say totalPendingSlots = 1000 and initTotalPending is 100 (which is current pending jobs in the system). We don't want to just submit just 20 percent in that case we want to submit up to the totalPendingSlots. 
I shouldn't add 20 percent extra in that case, but code will be more complicated and I think it is OK to overflow.

Reply:
From L543, isn't initTotalPending == totalPendingSlots ? Well, at least for the first time we iterate over a specific site. Actually, to me it seems it's always the same, since we don't do anything else with the pending slots (+1 or -1 etc) inside the submitter algo.

Reply:
Yes you're right. You found a bug. I was meant to put initiTotalPedning as initial totalPendingJobs not totalPendingSlots. Thanks I will update the patch.